File: samples/CS/05-Spinner/RibbonItems.cs
using System;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        public void Init()
        {

            ButtonDropA.Click += new EventHandler<EventArgs>(_buttonDropA_Click);
            Spinner.RepresentativeString = "2.50 m";
        }

        void _buttonDropA_Click(object sender, EventArgs e)
        {
            InitSpinner();
        }

        private void InitSpinner()
        {
            Spinner.DecimalPlaces = 2;
            Spinner.DecimalValue = 1.8M;
            Spinner.TooltipTitle = "Height";
            Spinner.TooltipDescription = "Enter height in meters.";
            Spinner.MaxValue = 2.5M;
            Spinner.MinValue = 0;
            Spinner.Increment = 0.01M;
            Spinner.FormatString = " m";
            Spinner.Label = "Height:";
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/05-Spinner/RibbonItems.vb
Imports System

Namespace WinForms.Ribbon

    Partial Class RibbonItems

        Public Sub Init()
			AddHandler ButtonDropA.Click, AddressOf _buttonDropA_ExecuteEvent
		End Sub

		Private Sub _buttonDropA_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			InitSpinner()
		End Sub

		Private Sub InitSpinner()
			_Spinner.DecimalPlaces = 2
			_Spinner.DecimalValue = 1.8D
			_Spinner.TooltipTitle = "Height"
			_Spinner.TooltipDescription = "Enter height in meters."
			_Spinner.MaxValue = 2.5D
			_Spinner.MinValue = 0
			_Spinner.Increment = 0.01D
			_Spinner.FormatString = " m"
			_Spinner.RepresentativeString = "2.50 m"
			_Spinner.Label = "Height:"
		End Sub

	End Class
End Namespace
